src/socket.rs

```diff
@@ -925,6 +925,48 @@ fn into_linger(duration: Option<Duration>) -> sys::linger {
 /// * Linux: <https://man7.org/linux/man-pages/man7/ip.7.html>
 /// * Windows: <https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options>
 impl Socket {
+    /// Get the value of the `IP_TRANSPARENTF` option on this socket.
+    ///
+    /// For more information about this option, see [`set_ip_transparent`].
+    ///
+    /// [`ip_transparent`]: Socket::set_ip_transparent
+    #[cfg(all(feature = "all", target_os = "linux"))]
+    pub fn ip_transparent(&self) -> io::Result<bool> {
+        unsafe {
+            getsockopt::<c_int>(self.inner, sys::IPPROTO_IP, libc::IP_TRANSPARENT)
```

Comment:
From the CI: can't use `self.inner` anymore, you need to use `self.as_raw()`.

Reply:
Forgive me, happy to change it, but this confuses me and I'd like to understand what's going on.

I don't see `as_raw` in the source code for this project, is that from somewhere else?  What does it do and why would my method need to be different than the others in the file?  They appear to be using the same function(s).

Changing self.inner --> self.as_raw() still compiles, but I don't really understand why.
It looks like `self.inner` is a `Socket` which is actually a `c_int` which is a `i32`.     
It also only seems to compile when used in this method, but put into other methods causes a ```no method named `as_raw` found for reference `&socket::Socket` in the current scope``` error.

Is there some implementation of `as_raw` for `Socket` or `c_int` somewhere?  
my rust-analyzer is choking on this file so thats not really helping me either.

Reply:
Does it have to do with the feature directive somehow?

Reply:
This is due to https://github.com/rust-lang/socket2/pull/222 which I merged yesterday, e.g. see https://github.com/rust-lang/socket2/blob/eaa5f8f66233df701983ec073f1c5683019e442e/src/socket.rs#L694-L697

It's the same as the old `.inner`, but we now use a `TcpStream` to take advantage of the niche representation of that type.

Reply:
Ok, rebased the branch against master and things look OK now.